docs/guide/styles.md

```diff
@@ -1,15 +1,320 @@
+
+```{.textual path="docs/examples/guide/styles/colors02.py"}
+```
+
+## Dimensions
+
+Widgets occupy a rectangular region of the screen, which may be as small as a single character or as large as the screen. Potentially *larger* if [scrolling](#) is enabled.
+
+### Box Model
+
+The following styles influence the dimensions of a widget.
+
+- [width](../styles/width.md) and [height](../styles/width.md) define the size of the widget.
+- [padding](../styles/padding.md) adds optional space around the content area. 
+- [border](../styles/border.md) draws an optional rectangular border around the padding and the content area.
+
+Additionally, the [margin](../styles/margin.md) style adds space around a widget's border, which isn't technically part of the widget, but provides visual separation between widgets.
+
+Together these styles compose the widget's *box model*. The following diagram shows how these settings are combined:
 
 <div class="excalidraw">
 --8<-- "docs/images/styles/box.excalidraw.svg"
 </div>
+
+### Width and height
+
+Setting the width restricts the number of columns used by a widget, and setting the height restricts the number of rows. Let's look at an example which sets both dimensions.
+
+```python title="dimensions01.py" hl_lines="21-22"
+--8<-- "docs/examples/guide/styles/dimensions01.py"
+```
+
+This code produces the following result.
+
+```{.textual path="docs/examples/guide/styles/dimensions01.py"}
+```
+
+Note how the text wraps in the widget, and is cropped because it doesn't fit in the space provided.
+
+#### Auto dimensions
+
+In practice, we generally want the size of a widget to adapt to it's content, which we can do by setting a dimension to `"auto"`.
+
+Let's set the height to auto and see what happens.
+
+```python title="dimensions02.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/dimensions02.py"
+```
+
+If you run this you will see the height of the widget now grows to accommodate the full text.:
+
+```{.textual path="docs/examples/guide/styles/dimensions02.py"}
+```
+
+#### Units
+
+Textual offers a few different *units* which allow you to specify dimensions relative to the screen or container. Relative units can better make use of available space if the user resizes the terminal.
+
+- Percentage units are given as a string containing a number followed by a percentage symbol, e.g. `"50%"`. Setting a dimension to a percentage unit will cause it to fit in that percentage of the available space. For instance, setting width to `"50%"` will cause the width of the widget to be half the available space.
+- View units are similar to percentage units, but explicitly reference a dimension. The `vw` unit sets a dimension to a percentage of the terminal *width*, and `vh` sets a dimension to a percentage of the terminal *height*.
+- The `w` unit sets a dimension to a percentage of the available width (which may be smaller than the terminal size if the widget is within another widget).
```

Comment:
Appreciate the text in the brackets here. I was like "but how does that differ from the bullet above" and my question was immediately answered!